lib/screens/map_view.dart

```diff
@@ -52,64 +52,75 @@ class _MapViewScreenState extends State<MapViewScreen> {
       }
 
       locationData = await locationService.getLocation();
-
     } on PlatformException catch (e) {
       debugPrint('Error: ${e.toString()}, code: ${e.code}');
       locationData = null;
     }
-    locationData = await locationService.getLocation();
     setState(() {});
   }
 
   Widget map(BuildContext context) {
+    MapControllerImpl mapController = MapControllerImpl();
     if (locationData == null) {
       return const Center(child: CircularProgressIndicator());
     } else {
-      return FlutterMap(
-        options: MapOptions(
-          center: latlng.LatLng(
-              locationData?.latitude ?? 51.5, locationData?.longitude ?? -0.09),
-          zoom: 18.0,
-          interactiveFlags: InteractiveFlag.none,
-        ),
-        layers: [
-          TileLayerOptions(
-              urlTemplate: mapboxURL, // ignore: undefined_identifier
-              attributionBuilder: (_) {
-                return const Text("© Mapbox");
-              },
-              additionalOptions: {
-                'accessToken': mapboxAPIKey, // ignore: undefined_identifier
-                'id': 'mapbox.mapbox-streets-v8'
-              }),
-          MarkerLayerOptions(
-            markers: [
-              Marker(
-                  width: 40.0,
-                  height: 40.0,
-                  point: latlng.LatLng(locationData?.latitude ?? 51.5,
-                      locationData?.longitude ?? -0.09),
-                  builder: (ctx) => const Beastie())
-            ],
-          ),
-        ],
-      );
+      return StreamBuilder(
```

Comment:
I might tinker with this at some point in the future to see if we actually need this or if we can just listen for location changes but for now this works so I think we can revisit that later if we have time.